packages/web-api/test/app.test.ts

```diff
@@ -1309,4 +1310,453 @@ describe('API', () => {
       });
     });
   });
+
+  describe('Pre-Approved Users', () => {
+    let preApprovedUserId: number;
+
+    beforeEach(async () => {
+      // Create a test pre-approved user
+      const preApproved = await prisma.preApprovedUser.create({
+        data: {
+          email: 'preapproved@example.com',
+          roles: ['ADMIN'],
+          created_by_user_id: user1Id
+        }
+      });
+      preApprovedUserId = preApproved.id;
+    });
+
+    describe('POST /api/auth/admin/pre-approved-users', () => {
+      it('should create a new pre-approved user', async () => {
+        const res = await authRequest(app, 'admin')
+          .post('/api/auth/admin/pre-approved-users')
+          .send({
+            email: 'newpreapproved@example.com',
+            roles: ['ADMIN']
+          })
+          .expect(201);
+
+        expect(res.body.preApprovedUser).toHaveProperty('email', 'newpreapproved@example.com');
```

Comment:
personal preference I guess, but I kinda prefer [toMatchObject](https://jestjs.io/docs/expect#tomatchobjectobject) for this kind of thing